### ISSUE TITLE
Refreshing the `tide-references` buffer

### ISSUE THREAD

**Issue by MarcusRiemer** [ananthakumaran/tide#157]
I am currently in the middle of obliterating an unused class and always have a `*tide-references*` buffer open. From what I can tell it is not possible to refresh that buffer without navigating to a source file, putting my cursor over the entity in question and calling `tide-references`. For my use case that means a lot of navigating back and forth because I am eliminating exactly the thing I would like to grab all references to.

It would be *really* great if I could refresh the buffer by just hitting `g` like in `dired`.

**Comment by josteink** [comment]
That sounds like a good improvement, if someone is willing to make it.

I think the biggest challenge would be finding out *how* and *what* to update based on.

Currently the way we get references, is that we call out to tsserver asking "What is pointing to this entity, at this particular point, in this particular file, whatever that may actually be?". We don't know *what* we're asking for.

So if the file in question in any way has been altered, asking the same question once again may return completely different results, for some completely different entity, and since we don't know what we asked for originally, we don't know how to re-locate it, so that we can ask for the same thing once again.

Basically, getting this working will take some effort.

If someone else is willing to do that, and make it work reliably, I'll be happy to merge though.

**Comment by MarcusRiemer** [comment]
Ah, so this is tightly coupled to the cursor position. That does indeed complicate things. I quickly glanced over this language server API and it doesn't seem to offer functionality to access references by some kind of canonical name, what a pity :(

I have never dived into emacs-lisp, but how hard would it be to associate the buffer with the exact location the query was made the last time and simply re-execute that call to `tide-references`? I am aware that this will go horribly wrong if I change the file I used for the original search. But in my narrow use case this would already help tremendously, as I could simply obliterate the actual implementation last.

Or to put in another way: I might want to get my feet wet with this, if at least the trivial implementation would be as trivial to make as it sounds in my head.

**Comment by MarcusRiemer** [comment]
I had another idea, that could actually help me out: As long as the `*tide-references*`-buffer still has existing references in it, its maybe possible to work around the problem by binding a key to an action like "please now call `tide-references` for the cursor position at this reference"?

This would also solve my immediate problem, but I have no idea whether this is actually easier to implement.

**Comment by josteink** [comment]
To be quite honest, I'm not sure what the best way to do this would be.

Right now we have some work to do in general with tide-references (like easier navigation between results). Maybe a feature like this should be considered too, when we have time to dig deeper into it.

Either way, if you want to experiment and get your hands dirty *now*, these are some relevant pieces of code:

* https://github.com/ananthakumaran/tide/blob/master/tide.el#L1123-L1134
* https://github.com/ananthakumaran/tide/blob/master/tide.el#L1021-L1025
* https://github.com/ananthakumaran/tide/blob/master/tide.el#L1027-L1034

That should hopefully be enough to get you started.

**Comment by MarcusRiemer** [comment]
Thanks a bunch, from the looks of it I wont be able to do this right now. But this does some like a nice thing to sink in an hour or two when its weekend.

**Comment by josteink** [comment]
Closing issue due to lack of activity. Feel free to re-open if you feel like reinvestigating this.
